source/array_table_model.ts

```diff
@@ -46,14 +69,28 @@ export class ArrayTableModel extends TableModel {
    * @throws RangeError - The source or destination are not within this table's
    *                      range.
    */
-  public move(source: number, destination: number): void {}
+  public move(source: number, destination: number): void {
+    if(source > this.rowCount - 1 || source < 0 ||
```

Comment:
Prefer ```x >= y``` over ```x > y - 1```. As a general principle it's preferable to use operation that best matches the output instead of manipulating the arguments to work with a worse matching operation.